applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.cpp

```diff
@@ -0,0 +1,524 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Fernando Rastellini
+//  Collaborators:   Riccardo Rossi
+//                   Vicente Mataix Ferrandiz
+//
+
+#include "plasticity_isotropic_kinematic_j2.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::PlasticityIsotropicKinematicJ2()
+    : ConstitutiveLaw()
+{
+}
+
+//********************************COPY CONSTRUCTOR************************************
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::PlasticityIsotropicKinematicJ2(const PlasticityIsotropicKinematicJ2 &rOther)
+    : ConstitutiveLaw(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer PlasticityIsotropicKinematicJ2::Clone() const
+{
+    return Kratos::make_shared<PlasticityIsotropicKinematicJ2>(*this);
+}
+
+//********************************DESTRUCTOR******************************************
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::~PlasticityIsotropicKinematicJ2()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool PlasticityIsotropicKinematicJ2::Has(const Variable<bool>& rThisVariable)
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        return true;
+    }
+    return false;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool& PlasticityIsotropicKinematicJ2::GetValue(
+    const Variable<bool>& rThisVariable,
+    bool& rValue
+    )
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        rValue = mPlasticEvolution;
+    }
+
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterial(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues
+    )
+{
+    mPlasticStrainLast = ZeroVector(this->GetStrainSize());
```

Comment:
Accordingly to our last discussion, there shouldn't be 4 member variables for 2 internal variables